lambdas/preprocessing/src/lambda_function.py

```diff
@@ -1,28 +1,75 @@
+import os
+from typing import Any, Dict, Optional
+from uuid import uuid4, UUID
+
+import geopandas as gpd
+import pandas as pd
+import tempfile
 from aws_xray_sdk.core import patch, xray_recorder
+from shapely.wkb import dumps as wkb_dumps
 
+from raster_analysis.boto import s3_client
 from raster_analysis.exceptions import QueryParseException
 from raster_analysis.globals import LOGGER
 
 patch(["boto3"])
 
+# FIXME: Get these from env
+BUCKET = "gfw-pipelines-test"
+REGION = "us-east-1"
+
 
 @xray_recorder.capture("Preprocessing")
 def handler(event, context):
     try:
         LOGGER.info(f"Running preprocessing with parameters: {event}")
+        fc: Optional[Dict] = event.get("feature_collection")
+        uri: Optional[str] = event.get("uri")
+        id_field = event.get("id_field", "fid")  # Reasonable to use a default?
+
+        if fc is not None and uri is not None:
+            raise Exception("Please specify GeoJSON via (only) one parameter!")
+        elif fc is not None:
+            gpdf = gpd.GeoDataFrame.from_features(fc, columns=[id_field])
+        elif uri is not None:
+            gpdf = gpd.read_file(uri, columns=[id_field])
+        else:
+            raise Exception("Please specify GeoJSON via (only) one parameter!")
+
+        rows = []
+        for record in gpdf.itertuples():
+            geom_wkb = wkb_dumps(getattr(record, "geometry"), hex=True)
+            rows.append([getattr(record, id_field), geom_wkb])
+
+        # FIXME: Hash those args for cacheability!
+        request_hash: UUID = uuid4()
+        geom_prefix = f"test/otf_lists/{str(request_hash)}/geometries.csv"
```

Comment:
I think we can make this more concrete now, since it's already in a test bucket.

I'd prefer not to include "otf" in the path name because it requires knowing an acronym we may or may not still use in 3 years. Maybe something general like `/analysis/jobs/input`?

Reply:
@jterry64 Sounds good, but where does the hash go? How about /analysis/jobs/$hash/input and /analysis/jobs/$hash/output ?